test/get-workspaces-paths-test.js

```diff
@@ -166,7 +166,10 @@ describe(function() {
         fixturify.writeSync(tmpPath, {
           'packages': {
             'package-a': {
-              'package.json': stringifyJson({}),
+              'package.json': stringifyJson({
+                'name': 'package-a',
```

Comment:
I see you added this for pnpm? but that shouldn't be correct. Pnpm doesn't have the same name and version requirement that yarn does. So maybe `processGlobs` needs to be different for each provider.

Reply:
or pass new function parameter to differentiate?